File: src/main/java/sprykerFileCreator/model/definition/spryker/ZedPersistenceFactoryDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ZedPersistenceFactoryDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.ZED_PERSISTENCE_FACTORY;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}PersistenceFactory";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Zed\\{{ bundleName }}\\Persistence";
    }

    @Override
    public String getMethodForReturnType() {
        return "getFactory()";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.ZED_CONFIG
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Zed\\Kernel\\Persistence\\AbstractPersistenceFactory";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/ZedPersistenceFactoryDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class ZedPersistenceFactoryDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.ZED_PERSISTENCE_FACTORY
    override val namePattern: String
        get() = "{{ bundleName }}PersistenceFactory"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Zed\\{{ bundleName }}\\Persistence"
    override val methodForReturnType: String
        get() = "getFactory()"
    override val docBlockClasses: Array<String>
        get() = arrayOf(
                SprykerConstants.ZED_CONFIG
        )
    override val defaultParentClass: String
        get() = "Spryker\\Zed\\Kernel\\Persistence\\AbstractPersistenceFactory"
}
